connect/src/protocols/tokenBridge/tokenTransfer.ts

```diff
@@ -167,9 +173,20 @@ export class TokenTransfer<N extends Network = Network>
     let from = { chain: vaa.emitterChain, address: vaa.emitterAddress };
     let { token, to } = vaa.payload;
 
+    let nativeAddress: NativeAddress<Chain>;
+    if (token.chain === from.chain) {
+      nativeAddress = await wh.getTokenNativeAddress(from.chain, token.chain, token.address);
+    } else {
+      const fromChain = await wh.getChain(from.chain);
+      const tb = await fromChain.getTokenBridge();
+      const wrapped = await tb.getWrappedAsset(token);
+      nativeAddress = toNative(token.chain, wrapped.toString());
+    }
+
+    const decimals = await wh.getDecimals(from.chain, nativeAddress);
     const scaledAmount = amount.scale(
-      amount.fromBaseUnits(token.amount, TokenTransfer.MAX_DECIMALS),
-      await wh.getDecimals(token.chain, token.address),
```

Comment:
Previously we were scaling by the token origin chain's decimals, but we should scale by the source chain's decimals which may be different (e.g., Wormhole-wrapped WETH on Solana has 8 decimals vs 18 decimals on Ethereum).